apis/python/src/tiledbsoma/io/registration/signatures.py

```diff
@@ -0,0 +1,305 @@
+            raw_var_schema=raw_var_schema,
+            obsm_dtypes=obsm_dtypes,
+            varm_dtypes=varm_dtypes,
+        )
+
+    @classmethod
+    def fromH5AD(
+        cls,
+        h5ad_file_name: str,
+        *,
+        default_obs_field_name: str = "obs_id",
+        default_var_field_name: str = "var_id",
+        default_X_layer_name: str = "data",
+    ) -> Self:
+        """
+        See ``fromAnnData``.
+        """
+        adata = ad.read_h5ad(h5ad_file_name, "r")
+        return cls.fromAnnData(adata, default_X_layer_name=default_X_layer_name)
+
+    @classmethod
+    def fromSOMAExperiment(cls, uri: str, measurement_name: str = "RNA") -> Self:
+        """
+        Constructs a pre-check signature from a SOMA experiment, which can be compared against
+        another signature from AnnData/H5AD or SOMA experiment.
+        """
+
+        with tiledbsoma.Experiment.open(uri) as exp:
+
+            obs_schema = _string_dict_from_arrow_schema(exp.obs.schema)
+
+            var_schema = _string_dict_from_arrow_schema(
+                exp.ms[measurement_name].var.schema
+            )
+
+            X_dtypes = {}
+            for X_layer_name in exp.ms[measurement_name].X.keys():
+                X = exp.ms[measurement_name].X[X_layer_name]
+                X_dtypes[X_layer_name] = str(X.schema.field("soma_data").type)
+
+            raw_X_dtype = None
+            raw_var_schema = None
+            if "raw" in exp.ms:
+                raw_var_schema = _string_dict_from_arrow_schema(
+                    exp.ms["raw"].var.schema
+                )
+
+                X = exp.ms["raw"].X[X_layer_name]
+                raw_X_dtype = str(X.schema.field("soma_data").type)
+
+            obsm_dtypes: Dict[str, str] = {}
+            obsm_dtypes = {}
+            if "obsm" in exp.ms[measurement_name]:
+                for obsm_layer_name in exp.ms[measurement_name].obsm.keys():
+                    obsm = exp.ms[measurement_name].obsm[obsm_layer_name]
+                    obsm_dtypes[obsm_layer_name] = str(
+                        obsm.schema.field("soma_data").type
+                    )
+
+            varm_dtypes: Dict[str, str] = {}
+            if "varm" in exp.ms[measurement_name]:
+                for varm_layer_name in exp.ms[measurement_name].varm.keys():
+                    varm = exp.ms[measurement_name].varm[varm_layer_name]
+                    varm_dtypes[varm_layer_name] = str(
+                        varm.schema.field("soma_data").type
+                    )
+
+            return cls(
+                obs_schema=obs_schema,
+                var_schema=var_schema,
+                X_dtypes=X_dtypes,
+                raw_X_dtype=raw_X_dtype,
+                raw_var_schema=raw_var_schema,
+                obsm_dtypes=obsm_dtypes,
+                varm_dtypes=varm_dtypes,
+            )
+
+    @classmethod
+    def compatible(cls, signatures: Dict[str, Self]) -> Tuple[bool, Optional[str]]:
```

Comment:
This seems more like you would want to handle it as a `check_compatibility` function that maybe raised an exception of some kind. The signature as-is has the risk of being misunderstood; a person could write

```python
if compatible(some_sigs):
  do_dangerous_thing(...)
```

and since a 2-tuple is non-empty, it would always evaluate as truthy and `do_dangerous_thing`